From 1fa672edd26c8c7745d31c4736c8d85b50894f79 Mon Sep 17 00:00:00 2001
From: Simeon Marijon <simeon.marijon@foss.st.com>
Date: Thu, 21 Sep 2023 09:39:09 +0200
Subject: [PATCH 0512/1695] dts: stm32: add reboot mode in stm32mp151.dtsi

reboot mode is a consumer of nvmem cell

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git bfce2995e6b2b95d860b90b0e5aacc63d5e7aff1 dts: stm32: add reboot mode in stm32mp151.dtsi]
Change-Id: I4afb361d7403e0ad0aaaa78cb6efbf4287c934a9
Signed-off-by: Simeon Marijon <simeon.marijon@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/328929
Tested-by: Simeon MARIJON <simeon.marijon@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Simeon MARIJON <simeon.marijon@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
(cherry picked from commit 949c8fcffc2f7ccdf593f9092918c14475025cd4)
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/351777
---
 core/arch/arm/dts/stm32mp151.dtsi | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp151.dtsi b/core/arch/arm/dts/stm32mp151.dtsi
index 368e2bca4..ae461cf82 100644
--- a/core/arch/arm/dts/stm32mp151.dtsi
+++ b/core/arch/arm/dts/stm32mp151.dtsi
@@ -1674,6 +1674,19 @@
 				};
 			};
 
+			reboot_mode: reboot-mode {
+				compatible = "nvmem-reboot-mode";
+				nvmem-cells = <&boot_mode>;
+				nvmem-cell-names = "reboot-mode";
+				mode-normal = <0x00>;
+				mode-fastboot = <0x01>;
+				mode-recovery = <0x02>;
+				mode-stm32cubeprogrammer = <0x03>;
+				mode-ums_mmc0 = <0x10>;
+				mode-ums_mmc1 = <0x11>;
+				mode-ums_mmc2 = <0x12>;
+				mode-romcode_serial = <0xff>;
+			};
 		};
 
 		/*
-- 
2.47.2

